src/shared/watchedFiles.ts

```diff
@@ -133,6 +162,24 @@ export abstract class WatchedFiles<T> implements vscode.Disposable {
         }
     }
 
+    /**
+     * Load a virtual document (e.g. untitled documents) into the registry by their content
+     * @param uri the uri of the virtual document
+     * @param contents the contents of the virtual document
+     */
+    public async addVirtualItemToRegistry(uri: vscode.Uri, contents: string): Promise<void> {
```

Comment:
I mostly did it this way to avoid having to pass through the contents with something like:
```
addItemToRegistry(uri: vscode.Uri, quiet?: boolean, contents?: string)
```

I'll play around with it and see if I can simplify some things

Reply:
Two methods is ok. The bigger problem is that we are distinguishing 'virtual' and 'non-virtual' things. Though this is probably because `load` is an abstract method accepting a path when it should be a method called `parse` accepting the file contents.